### PR TITLE
fix: Evm Can't transfer or burn your own NFT

### LINKED ISSUE #11219: Evm Can't transfer or burn your own NFT
### What happened?

CleanShot 2024-12-07 at 07 34 07@2x

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Closes #11219

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

### Sub

CleanShot 2024-12-08 at 06 46 18@2x

### Evm

CleanShot 2024-12-08 at 06 43 55@2x

### CODE REVIEW

File: utils/account.ts [hunk 1]
```diff
@@ -69,13 +69,24 @@ export const formatAccount = (
...
   account2: KeyringAccount | string,
 ): boolean => {
   const address1 = accountToAddress(account1)
   const address2 = accountToAddress(account2)
-  return addressEq(address1, address2)
+
+  return isEthereumAddress(address1)
+    ? isSameAddressEvm(address1, address2)
+    : isSameAddressSubstrate(address1, address2)
 }
```
Comment (preschian) [inline utils/account.ts:90]: can we solve it by this?

```suggestion
  return addressEq(address1.toLowerCase(), address2.toLowerCase())
```

**Review by preschian — APPROVED** [review]

File: utils/account.ts (hunk 1 above)
Comment (hassnian) [reply utils/account.ts:90]: nope 

CleanShot 2024-12-11 at 07 22 46@2x

**Comment by vikiival** [comment]
cc @Jarsen136 for review
cc @hassnian for conflicts

**Review by Jarsen136 — APPROVED** [review]
lgtm

**Pull request merged**